burn-import/src/burn/node/constant.rs

```diff
@@ -1,72 +1,153 @@
 use super::{Node, NodeCodegen};
-use crate::burn::{OtherType, Scope, Type};
-use burn::record::PrecisionSettings;
+use crate::burn::{ScalarKind, ScalarType, Scope, TensorType, ToTokens, Type};
+use burn::{record::PrecisionSettings, tensor::DataSerialize};
 use proc_macro2::{Ident, Span, TokenStream};
 use quote::quote;
+use serde::Serialize;
 
 #[derive(Debug, Clone)]
-pub struct ConstantNode {
+pub struct ConstantNode<PS: PrecisionSettings> {
     pub name: String,
-    pub value: ConstantValue,
-    output_ty: OtherType,
+    pub value: ConstantValue<PS>,
+    pub output: Type,
+}
+
+#[derive(Debug, Clone)]
+pub enum TensorValue<PS: PrecisionSettings> {
+    Float(DataSerialize<PS::FloatElem>),
+    Int(DataSerialize<PS::IntElem>),
 }
 
 #[derive(Debug, Clone, new)]
-pub enum ConstantValue {
-    Int(i32),
-    Float(f32),
-    Bool(bool),
+pub enum ConstantValue<PS: PrecisionSettings> {
+    /// Float constant.
+    Float32(f32),
+
+    /// Integer constant.
+    Int64(i64),
+
+    /// Tensor constant (does not hold a value).
+    Tensor(TensorType, TensorValue<PS>),
```

Comment:
This is a bit confusing, the comment says does not hold a value but there is `TensorValue<PS>`.

Reply:
Outdated comment. I'll remove it.